.upstream-tests/test/cuda/atomics/atomic.ext/atomic_fetch_max.pass.cpp

```diff
@@ -60,14 +60,58 @@ struct TestFn {
   }
 };
 
+template <template<typename, typename> typename Selector, cuda::thread_scope ThreadScope>
+struct TestFn<int, Selector, ThreadScope> {
+  __host__ __device__
+  void operator()() const {
+    // Test greater
+    {
+        typedef cuda::atomic<int> A;
+        Selector<A, constructor_initializer> sel;
+        A & t = *sel.construct();
+        t = int(1);
+        assert(t.fetch_max(2) == int(1));
+        assert(t.load() == int(2));
+    }
+    {
+        typedef cuda::atomic<int> A;
+        Selector<volatile A, constructor_initializer> sel;
+        volatile A & t = *sel.construct();
+        t = int(1);
+        assert(t.fetch_max(2) == int(1));
+        assert(t.load() == int(2));
+    }
+    // Test not greater
+    {
+        typedef cuda::atomic<int> A;
+        Selector<A, constructor_initializer> sel;
+        A & t = *sel.construct();
+        t = int(-1);
+        assert(t.fetch_max(4) == int(-1));
```

Comment:
This does not connect with the comment. 

We are still testing a smaller `int` versus a larger threshold. Why is this changed and could we update the comment? 

If this does some horrible conversion to unsigned magic we should test that explicitly and keep the basic 3 vs 2 test

Reply:
There's no horrible conversion, it's just specifically testing int types. The cast is, in truth, unnecessary.